packages/adapter/src/methods.ts

```diff
@@ -11,6 +14,26 @@ export async function getAccountAddress(pluginOrigin: string): Promise<string> {
   }) as string;
 }
 
+async function _sign(payload: SignerPayloadJSON|SignerPayloadRaw, paramKey: "payloadJSON"|"payloadRaw") {
+  return await window.ethereum.send({
+    method: pluginOrigin,
+    params: [{
+      method: 'signPayload',
+      params: {
+        [paramKey]: payload
```

Comment:
I think payload has toJson method, and we should definetly serialize it to avoid metamask issues with bigint and stuff

Reply:
@mpetrunic there is no `toJson` method, but if you look at definitions of `SignerPayloadJSON` and `SignerPayloadRaw` below you can see that there are no problematic properties. I can serialize it here to JSON string and attach it as a `payload` parameter then?

```js
export interface SignerPayloadJSON {
    /**
     * @description The ss-58 encoded address
     */
    address: string;
    /**
     * @description The checkpoint hash of the block, in hex
     */
    blockHash: string;
    /**
     * @description The checkpoint block number, in hex
     */
    blockNumber: string;
    /**
     * @description The era for this transaction, in hex
     */
    era: string;
    /**
     * @description The genesis hash of the chain, in hex
     */
    genesisHash: string;
    /**
     * @description The encoded method (with arguments) in hex
     */
    method: string;
    /**
     * @description The nonce for this transaction, in hex
     */
    nonce: string;
    /**
     * @description The current spec version for  the runtime
     */
    specVersion: string;
    /**
     * @description The tip for this transaction, in hex
     */
    tip: string;
    /**
     * @description The version of the extrinsic we are dealing with
     */
    version: number;
}
```

```js
export interface SignerPayloadRawBase {
    /**
     * @description The hex-encoded data for this request
     */
    data: string;
    /**
     * @description The type of the contained data
     */
    type?: 'bytes' | 'payload';
}
export interface SignerPayloadRaw extends SignerPayloadRawBase {
    /**
     * @description The ss-58 encoded address
     */
    address: string;
    /**
     * @description The type of the contained data
     */
    type: 'bytes' | 'payload';
}
```

Reply:
Then it's fine as is, not sure where I saw toJson